stdlib/internal/SwiftSyntax/SyntaxChildren.swift

```diff
@@ -0,0 +1,28 @@
+//===------------- SyntaxChildren.swift - Syntax Child Iterator -----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// A sequence that allows iteration of the children of a Syntax node.
+public struct SyntaxChildren: Sequence {
+  /// The node being iterated.
+  let node: Syntax
+
+  /// Creates an iterator over the chilren of the provided node.
+  public func makeIterator() -> AnyIterator<Syntax> {
+    var index = 0
+    return AnyIterator {
```

Comment:
Oh, you don't want to do this ;-) (I looked inside `AnyIterator` and I saw things in there). Writing a special `Iterator` struct for `SyntaxChildren` is just a few more lines of code, but avoids an unnecessary allocation.